src/sentry/api/serializers/models/group.py

```diff
@@ -682,14 +686,19 @@ def _is_authorized(user, organization_id: int):
         if (
             request
             and getattr(request.user, "is_sentry_app", False)
-            and isinstance(request.auth, ApiToken)
+            and is_api_token_auth(request.auth)
         ):
             if SentryAppInstallationToken.objects.has_organization_access(
                 request.auth, organization_id
             ):
                 return True
 
-        return user.is_authenticated and user.get_orgs().filter(id=organization_id).exists()
+        return (
+            user.is_authenticated
+            and OrganizationMember.objects.filter(
+                user_id=user.id, organization_id=organization_id
```

Comment:
We need to add a filter here for visibility status to replicate the original behavior, adding.

Reply:
Actually, never mind, that filter is not effective for this due to the `filter`.